core-bundle/src/DataContainer/DataContainerGlobalOperationsBuilder.php

```diff
@@ -0,0 +1,269 @@
+        ]);
+
+        return $this;
+    }
+
+    public function addGlobalButtons(DataContainer $dataContainer, callable|null $legacyCallback = null): self
+    {
+        $this->ensureInitialized();
+
+        if (!\is_array($GLOBALS['TL_DCA'][$this->table]['list']['global_operations'] ?? null)) {
+            return $this;
+        }
+
+        $inputAdapter = $this->framework->getAdapter(Input::class);
+
+        foreach ($GLOBALS['TL_DCA'][$this->table]['list']['global_operations'] as $k => $v) {
+            if (!($v['showOnSelect'] ?? null) && 'select' === $inputAdapter->get('act')) {
+                continue;
+            }
+
+            $v = \is_array($v) ? $v : [$v];
+            $operation = $this->generateOperation($k, $v, $dataContainer, $legacyCallback);
+
+            if ($operation) {
+                $this->append($operation);
+            }
+        }
+
+        return $this;
+    }
+
+    private function generateOperation(string $name, array $operation, DataContainer $dataContainer, callable|null $legacyCallback = null): array|null
+    {
+        $config = new DataContainerOperation($name, $operation, null, $dataContainer);
+
+        // Call a custom function instead of using the default button
+        if (\is_array($operation['button_callback'] ?? null)) {
+            $callback = $this->framework->getAdapter(System::class)->importStatic($operation['button_callback'][0]);
+            $ref = new \ReflectionMethod($callback, $operation['button_callback'][1]);
+
+            if (
+                1 === $ref->getNumberOfParameters()
+                && ($type = $ref->getParameters()[0]->getType())
+                && $type instanceof \ReflectionNamedType
+                && DataContainerOperation::class === $type->getName()
+            ) {
+                $callback->{$operation['button_callback'][1]}($config);
+            } else {
+                if (!$legacyCallback) {
+                    throw new \RuntimeException('Cannot handle legacy button_callback, provide the $legacyCallback');
+                }
+
+                $legacyCallback($config);
+            }
+        } elseif (\is_callable($operation['button_callback'] ?? null)) {
+            $ref = new \ReflectionFunction($operation['button_callback']);
+
+            if (
+                1 === $ref->getNumberOfParameters()
+                && ($type = $ref->getParameters()[0]->getType())
+                && $type instanceof \ReflectionNamedType
+                && DataContainerOperation::class === $type->getName()
+            ) {
+                $operation['button_callback']($config);
+            } else {
+                if (!$legacyCallback) {
+                    throw new \RuntimeException('Cannot handle legacy button_callback, provide the $legacyCallback');
+                }
+
+                $legacyCallback($config);
+            }
```

Comment:
```suggestion
            } elseif ($legacyCallback) {
                $legacyCallback($config);
            } else {
                throw new \RuntimeException('Cannot handle legacy button_callback, provide the $legacyCallback');
            }
```

Same as above, feel free to ignore but feels more readable